README.md

```diff
@@ -1,18 +1,128 @@
+
+ <a href="https://github.com/aave-starknet-project/aave-starknet-bridge/blob/main/LICENSE.md"><img src="https://img.shields.io/github/license/sourcerer-io/hall-of-fame.svg?colorB=ff0000"></a>
+
+:warning: This codebase is still in an experimental phase, has not been audited, might contain bugs and should not be used in production.
+
+## Overview
+
+For Aave, one of the main current and future goals is growth of liquidity and user base. As seen on side-chains with low transaction cost like Polygon or Avalanche, there is high demand to use the Aave protocol with small amounts to earn high yield.
+
+Both previous points show that having an initial phase on the Aave <> Starknet integration allowing deposit/withdrawal on Aave Ethereum by exclusively transacting on Starknet can be a good idea, target-wise.
+
+The bridge allows users to deposit and withdraw `staticATokens` - wrappers converting balance-increasing [aTokens]( https://docs.aave.com/developers/tokens/atoken) into exchange-rate-increasing staticATokens - on StarkNet and get wrapped tokens `ETHStaticATokens` that allow users to keep enjoying the same rewards as on L1. 
+
+The bridge was also shaped for liquidity providers who are able to assume Ethereum cost of deposit/withdrawal, as they transact amounts big enough. They will deposit on Aave Ethereum, bridge the staticATokens to Starknet and make them available for users there to buy and hold, accruing this way yield from L1. 
 
 ## Architecture
 
-![Starknet AAVE rewards architectural diagram](resources/architecture.png)
-## Testing
+![aave_bridge_5](https://user-images.githubusercontent.com/37840702/165796331-f587b34c-74a4-4954-a05c-bf55705963d6.png)
 
-The project is tested using [hardhat](https://hardhat.org/), the [starknet
-hardhat plugin](https://github.com/Shard-Labs/starknet-hardhat-plugin) and
-[starknet-devnet](https://github.com/Shard-Labs/starknet-devnet).
+## Contracts
+
+`L1`
+  * `StaticATokenLM` - an updated implementation of staticATokens which makes it possible to communicate with its respective L2 token by sending the latest `rewards_index` on token transfer or when explicitly triggered to do so.
+  *  `TokenBridge` -  handles rewards update, deposit & withdrawal of staticATokens & their underlying assets
+  *  `Proxy` - A proxy implementation 
+
+`L2`
+  * `ETHStaticAToken` - Tokens on Starknet equivalent to each staticAToken on Ethereum mainnet. Contains the same logic for tracking user rewards as staticATokens and has the same `rewards_index`.
+  * `claimable` - tracks users' claimable rewards and current reward index for each `ETHStaticAToken`
+  * `token_bridge` - is responsible for:
+    * bridging the staticATokens to and from L2. Minting and burning ETHStaticATokens on message from L1. 
+    * bridging rewAAVE token back to L1
+    * updating `rewards_index` for each ETHStaticAToken on message from L1 
+  * `rewAAVE` - a very basic ERC20 to represent the rewards on L2
```

Comment:
Maybe we should leave out 'a very basic' and instead just say:
'an ERC20 representing the rewards on L2'